source/slang/slang-compiler.cpp

```diff
@@ -2536,12 +2536,27 @@ namespace Slang
                 if (allTargetsCUDARelated && targets.getCount() > 0)
                     continue;
 
-                auto numThreadsAttr = funcDecl->findModifier<NumThreadsAttribute>();
-                if (numThreadsAttr)
-                    profile.setStage(Stage::Compute);
-                else
+                bool canDetermineStage = false;
+                for (auto modifier : funcDecl->modifiers)
+                {
+                    if (as<NumThreadsAttribute>(modifier))
+                    {
+                        if (funcDecl->findModifier<OutputTopologyAttribute>())
+                            profile.setStage(Stage::Mesh);
```

Comment:
When OutputTopology is used together with numthreads, it can only be a mesh shader. Hull shader can also use outputtopology, but numthreads isn't valid there.